guide/src/module.md

```diff
@@ -138,47 +122,40 @@ mod my_extension {
     #[pymodule]
     mod submodule {
         // This is a submodule
-    }
-
-    #[pymodule_init]
-    fn init(m: &Bound<'_, PyModule>) -> PyResult<()> {
-        // Arbitrary code to run at the module initialization
-        m.add("double2", m.getattr("double")?)
+        use pyo3::prelude::*;
+                
+        #[pyclass]
+        struct Nested;
     }
 }
 # }
 ```
 
-The `#[pymodule]` macro automatically sets the `module` attribute of the `#[pyclass]` macros declared inside of it with its name.
+In this case, `#[pymodule]` macro automatically sets the `module` attribute of the `#[pyclass]` macros declared inside of it with its name.
 For nested modules, the name of the parent module is automatically added.
-In the following example, the `Unit` class will have for `module` `my_extension.submodule` because it is properly nested
-but the `Ext` class will have for `module` the default `builtins` because it not nested.
+In the previous example, the `Nested` class will have for `module` `my_extension.submodule`.
 
-```rust,no_run
-# mod declarative_module_module_attr_test {
-use pyo3::prelude::*;
+You can provide the `submodule` argument to `#[pymodule()]` for modules that are not top-level modules in order for them to properly generate the `#[pyclass]` `module` attribute automatically.
```

Comment:
I think this probably needs to be moved up into the submodules section (and can have a note saying the inline form will do it automatically).